activerecord/lib/active_record/query_logs_formatter.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module QueryLogs
+    class Formatter # :nodoc:
+      attr_reader :key_value_separator
+
+      # @param [String] key_value_separator: indicates the string used for
+      # separating keys and values.
+      #
+      # @param [Symbol] quote_values: indicates how values will be formatted (eg:
+      # in single quotes, not quoted at all, etc)
+      def initialize(key_value_separator:)
+        @key_value_separator = key_value_separator
+      end
+
+      # @param [String-coercible] value
+      # @return [String] The formatted value that will be used in our key-value
+      # pairs.
+      def format_value(value)
+        value
+      end
+    end
+
+    class QuotingFormatter < Formatter # :nodoc:
+      def format_value(value)
+        "'#{value.to_s.gsub("'", "\\\\'")}'"
```

Comment:
Note that I discovered one edge case since the last review of this implementation - if `value` isn't a string, it'll cause a runtime error unless we call `value.to_s` here.

An example is when someone passes a value like `Process.pid` (which we're doing in the tests below). I'm on the fence about it, but decided to coerce into a string because it's more user-friendly, and consistent with the legacy implementation (see related comment).

We can easily remove this `.to_s` call if we want to enforce that `value` must always be a string:
```suggestion
        "'#{value.gsub("'", "\\\\'")}'"
```
I'm fine with either approach :)